### ISSUE TITLE
scanpy fails to install with python 3.10

### ISSUE THREAD

**Issue by rmandla** [scverse/scanpy#2105]
Hi,

I am having trouble installing scanpy on 5.12 Manjaro with Python 3.10. I believe it is because llvmlite currently [does not support python 3.10](https://github.com/numba/llvmlite/issues/804#issuecomment-1002971267). Is there a way I can install scanpy with an older version of llvmlite? 

```python
pip install --user scanpy
```

```pytb

ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/setup.py'"'"'; __file__='"'"'/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-j38v6hmh/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/ube/.local/include/python3.10/llvmlite
         cwd: /tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/
    Complete output (29 lines):
    running install
    running build
    got version from file /tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/llvmlite/_version.py {'version': '0.34.0', 'full': 'c5889c9e98c6b19d5d85ebdd982d64a03931f8e2'}
    running build_ext
    /usr/bin/python /tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/ffi/build.py
    LLVM version... Traceback (most recent call last):
      File "/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/ffi/build.py", line 105, in main_posix
        out = subprocess.check_output([llvm_config, '--version'])
      File "/usr/lib/python3.10/subprocess.py", line 420, in check_output
        return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
      File "/usr/lib/python3.10/subprocess.py", line 501, in run
        with Popen(*popenargs, **kwargs) as process:
      File "/usr/lib/python3.10/subprocess.py", line 966, in __init__
        self._execute_child(args, executable, preexec_fn, close_fds,
      File "/usr/lib/python3.10/subprocess.py", line 1842, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'llvm-config'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/ffi/build.py", line 191, in <module>
        main()
      File "/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/ffi/build.py", line 181, in main
        main_posix('linux', '.so')
      File "/tmp/pip-install-i_hyvl0s/llvmlite_d91917e9522a491da51d00bc9034d43e/ffi/build.py", line 107, in main_posix
        raise RuntimeError("%s failed executing, please point LLVM_CONFIG "
    RuntimeError: llvm-config failed executing, please point LLVM_CONFIG to the path for llvm-config
    error: command '/usr/bin/python' failed with exit code 1
    ----------------------------------------
```

Thanks!

**Comment by ivirshup** [comment]
The upper bound for python versions we support is set by `numba` and `llvm-lite`. These rely on python internals, so can take a bit to be compatible with new releases.

There should be python 3.10 support in the next numba release (https://github.com/numba/numba/issues/7562), which has a candidate out now ([see details here](https://numba.discourse.group/t/numba-0-55-0-rc1/1075)).

I was able to get scanpy to import by doing:

```sh
mamba create -yn "numba-0.55.0rc1" "python=3.10"
conda activate numba-0.55.0rc1
mamba install -c numba -c numba/label/numpy numba=0.55.0rc1 numpy=1.21
pip install scanpy
python -c "import scanpy"
```

I couldn't install `scanpy` through conda since `pytables` doesn't have a 3.10 conda build afaict.

Please let me know if try this out and run into any issues!

**Comment by ivirshup** [comment]
I was able to set up a dev environment with a little work. I think it works? Ran into some other issues though.

<details>
<summary> Roughly what I ran </summary>

```sh
mamba create -yn "numba-0.55.0rc1-pip" -c conda-forge python=3.10 pip
conda activate numba-0.55.0rc1-pip
pip install "numba== 0.55.0rc1-pip"
pip install flit_core setuptools_scm
cd ~/github/anndata
pip install -e ".[dev,doc,test]"
cd ~/github/scanpy
pip install -e --no-build-isolation ".[dev,doc,test]"
```

</details>

**Comment by ivirshup** [comment]
Should be solved not that numba 0.55.0 is out.
